src/datalink/mod.rs

```diff
@@ -58,6 +59,9 @@ pub enum ChannelType {
     Layer3(EtherType),
 }
 
+/// Type of timestamped Ethernet packet (Linux only)
+pub struct TimestampedEthernetPacket<'a>(Duration, EthernetPacket<'a>);
```

Comment:
The macro raised errors when I used a tuple. Or are you suggesting `type EthernetPacketTimestamped<'a> = (Duration, EthernetPacket<'a>)`, which can then be used with `let (ts, pkt) = iter.next()`?